src/Backend/Core/Engine/Model.php

```diff
@@ -198,18 +198,24 @@ public static function deleteExtra($module = null, $type = null, array $data = n
 
         // loop found extras
         foreach ($extras as $extra) {
+            $deleteExtra = true;
+
             // match by parameters
             if ($data !== null && $extra['data'] !== null) {
                 $extraData = (array) unserialize($extra['data']);
 
-                // skip extra if parameters do not match
-                if (count(array_intersect($data, $extraData)) !== count($data)) {
-                    continue;
+                // do not delete extra if parameters do not match
+                foreach ($data as $dataKey => $dataValue) {
```

Comment:
This is possible using http://php.net/manual/en/function.array-diff-assoc.php. This is probably more efficient